File: Java/String-2/countHi.java
/*
 * Return the number of times that the string "hi" appears
 * anywhere in the given string.
 */

public int countHi(String str) {
  int count = 0;
  // Check how many times "hi" is said in the string.
  for(int i = 0; i < str.length() - 1; i++){
    if(str.substring(i, i + 2).equals("hi"))
      count++;
  }
  return count;
}

File: Python/String-2/count_hi.py
# Return the number of times that the string "hi"
# appears anywhere in the given string.

def count_hi(str):
  count = 0
  
  # Check to see how many times 'hi' appears in a given string
  for i in range(len(str) - 1):
    if (str[i:i + 2] == 'hi'):
      count += 1
  
  return count
